c/datasetjson.c

```diff
@@ -1630,6 +1640,395 @@ void respondWithDataset(HttpResponse* response, char* absolutePath, int jsonMode
+              respondWithError(response, HTTP_STATUS_INTERNAL_SERVER_ERROR, "Failed to set semaphore");
+              return;                  
+            }
+          else
+            {
+              zowelog(NULL, LOG_COMP_DATASERVICE, ZOWE_LOG_DEBUG,
+                "ZSS2020I %s semaphore %d was set\n", __FUNCTION__, semaphoreID);
+            }
+
+          /* the HTTP GET that consumes this message insists that the response body be JSON */
+          respondWithMessage(response, HTTP_STATUS_OK,
+                     "{\"records\":[\"Enqueue dataset successful\"]}"
+                     ); 
+
+          /********/
+          /* wait */
+          /********/
+          /* WAIT! */
+          zowelog(NULL, LOG_COMP_DATASERVICE, ZOWE_LOG_DEBUG,
+                  "ZSS1916I %s wait on semaphore %d, PID %d\n", __FUNCTION__, semaphoreID, pid);
+
+          /* define semaphore operation to be performed */
+          struct sembuf semaphoreBuffer[1]; /* just one semaphore in the array */
+          struct sembuf *semaphoreOps = &semaphoreBuffer[0];
+          semaphoreBuffer[0].sem_num = 0; /* index of first and only semaphore */
+          semaphoreBuffer[0].sem_op  = 0; /* 0 = wait */
+          semaphoreBuffer[0].sem_flg = 0; /* 0 = sychronous + don't undo */
+
+          /* wait for semaphore to be posted */
+          
+          semaphoreRetcode = semop(semaphoreID, semaphoreOps, 1); /* 1 = one entry */
+                /* we are now waiting for our semaphore to be posted ... */
+
+          
+          if (semaphoreRetcode != 0 )
+          {
+              zowelog(NULL, LOG_COMP_DATASERVICE, ZOWE_LOG_DEBUG,
+                "ZSS1938I %s Failed to wait on semaphore, retcode = %d\n", __FUNCTION__, semaphoreRetcode);
+          }
+          else
+          {
+              zowelog(NULL, LOG_COMP_DATASERVICE, ZOWE_LOG_DEBUG,
+                  "ZSS1931I %s semaphore %d has been posted, PID %d\n", __FUNCTION__, semaphoreID, pid);
+              /* destroy our semaphore */
+              semaphoreRetcode = semctl(semaphoreID, 0, IPC_RMID);
+              if (semaphoreRetcode == -1 )
+              {
+                  printf("ZSS1955I %s Failed to destroy semaphore\n", __FUNCTION__);}
+              else
+              {
+                  printf("ZSS1958I %s Destroyed semaphore %d, PID %d\n", __FUNCTION__, semaphoreID, pid);
+                  sem_table_entry[i].sem_ID = 0;  /* mark as deleted */
+              }
+          }
+         
+        } /* end of got semaphore */
+      }
+      else
+      { /* , do nothing if already present */
+        zowelog(NULL, LOG_COMP_DATASERVICE, ZOWE_LOG_DEBUG,
+            "ZSS1997I %s semaphore %d is already present \n", __FUNCTION__, semaphoreID);
+        respondWithError(response, HTTP_STATUS_RESOURCE_CONFLICT , "dataset is already enqueued"); 
+      } /* end of create semaphore */
+  /* end of semaphore section */
+
+} /* end of respondWithEnqueue */
+
+
+/* new function for DEQ */
```

Comment:
Rather than stating the obvious, it's better to provide a real description of the purpose and high level logic.